feat(bandwidth): add global bandwidth limiting for WireGuard interface

aaron-tsar opened the pull request:
---

## Description

This pull request adds a **global bandwidth limiting feature** for WireGuard VPN traffic using Linux Traffic Control (`tc`) with the HTB qdisc.

Bandwidth limits are applied at the **WireGuard interface level**, meaning all connected clients share the same configured limit. Both **download and upload** bandwidth can be configured (in Mbps) via the Admin UI. Upload limiting is supported through an **IFB (Intermediate Functional Block)** device, with a graceful fallback when the IFB kernel module is unavailable.

The configuration is portable and persistent, with settings stored in a SQLite database and preserved across container restarts via volume mounts.

---

## Motivation and Context

Currently, WireGuard traffic is unrestricted, which can lead to bandwidth abuse or uneven resource usage across VPN clients.

This feature allows administrators to:

* Control overall VPN bandwidth usage
* Prevent network saturation
* Configure limits easily via the UI
* Retain settings across restarts and deployments
* Run safely in containerized environments with graceful degradation

---

## How has this been tested?

* Enabled bandwidth limiting from **Admin > General**
* Configured download and upload limits (e.g. 10 Mbps)
* Verified `tc` rules were applied:

  ```
  tc qdisc show dev wg0
  ```
* Performed speed tests through the VPN to confirm throttling
* Restarted the container and confirmed settings persisted
* Tested upload limiting with IFB kernel module available
* Verified warning behavior when IFB module was unavailable and upload limiting was disabled gracefully

---

## Screenshots (if appropriate)

A new **Bandwidth Settings** section is available in **Admin > General**, containing:

* Enable/disable bandwidth limiting toggle
* Download limit input (Mbps)
* Upload limit input (Mbps)
<img width="1235" height="628" alt="image" src="https://github.com/user-attachments/assets/e5060d03-d222-4c6a-b6b6-9513cd13561e" />

---

## Types of changes

* [ ] Bug fix (non-breaking change which fixes an issue)
* [x] New feature (non-breaking change which adds functionality)
* [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

---

## Checklist

### General

* [x] Code follows the existing project coding standards and conventions
* [x] Feature is backward compatible and non-breaking
* [x] No existing functionality is affected by this change
* [x] Errors are handled gracefully and logged appropriately

### Database

* [x] Database migration `0003_bandwidth_limit.sql` is included
* [x] Migration is safe to run on existing installations
* [x] Bandwidth configuration persists using SQLite with volume mounts

### Backend

* [x] Bandwidth limiting is implemented using Linux `tc` with HTB qdisc
* [x] Upload limiting uses IFB device when available
* [x] Graceful fallback is implemented when IFB kernel module is unavailable
* [x] `tc` rules are applied and removed via WireGuard `PostUp` / `PostDown` hooks
* [x] New `/api/admin/bandwidth-status` endpoint returns correct bandwidth and IFB status
* [x] Failures in `tc` command execution do not crash the application

### Frontend

* [x] Bandwidth settings section is available in **Admin > General**
* [x] Enable/disable toggle correctly applies and removes bandwidth limits
* [x] Download and upload limit inputs validate numeric Mbps values
* [x] IFB availability is reflected correctly in the UI
* [x] Warning is shown when upload limiting is unavailable
* [x] UI state persists correctly after page reload

### i18n

* [x] All new strings are translated across all 16 supported locales
* [x] No missing or fallback translation keys remain

### Docker / Deployment

* [x] `iproute2` package is included in the Docker image
* [x] `tc` command is available inside the container
* [x] Feature works correctly after container restart

### Testing

* [x] Bandwidth limiting can be enabled and disabled without restarting the service
* [x] Download bandwidth throttling works as expected
* [x] Upload bandwidth throttling works when IFB module is available
* [x] Upload limiting is skipped gracefully when IFB is unavailable
* [x] Settings persist across restarts

### Documentation

* [x] Admin documentation is updated to describe the bandwidth limiting feature
* [x] Limitations and IFB requirements are clearly documented

---

kaaax0815 reviewed (changes requested):
remove ai leftovers,
dont translate using ai

aaron-tsar committed:
feat: add global bandwidth limiting for WireGuard VPN traffic

Add ability to limit download/upload speeds for all VPN clients using
Linux Traffic Control (tc) with HTB qdisc.

Features:
- Admin panel UI to enable/disable and configure bandwidth limits
- Download limiting via tc on WireGuard interface egress
- Upload limiting via IFB device mirroring ingress traffic
- Auto-detection of IFB kernel module availability
- Auto-restart interface when bandwidth settings change

Technical:
- New database migration for bandwidth settings
- tc commands injected as PostUp/PostDown hooks
- API endpoints: GET/POST /api/admin/general, GET /api/admin/bandwidth-status

i18n: Add translations for 16 languages
docs: Add bandwidth limiting setup guide

aaron-tsar commented:
> remove ai leftovers, dont translate using ai

Hi, I’ve pushed a new commit fixing the translations and removing AI references. 
Please take another look. Thanks!

kaaax0815 reviewed (changes requested):
you did not remove all the translations for languages you dont speak. ai translations are generally pretty bad

aaron-tsar committed:
fix: address PR review comments

- Remove Docker Container requirements (already in template)
- Remove Technical Details section (belongs in code only)
- Remove AI-generated translations (keep English only)

aaron-tsar committed:
refactor(bandwidth): move validation to Zod and simplify tc commands

aaron-tsar committed:
Merge branch 'master' into feat/bandwidth-limiting

aaron-tsar committed:
fix: correct migration timestamp order for 0003_bandwidth_limit

aaron-tsar committed:
ci: add build workflow for fork

aaron-tsar committed:
chore: remove fork build workflow

aaron-tsar committed:
Merge branch 'master' into feat/bandwidth-limiting

aaron-tsar committed:
Merge branch 'master' into feat/bandwidth-limiting
